Src/Newtonsoft.Json/Utilities/LinqBridge.cs

```diff
@@ -151,11 +158,20 @@ private static IEnumerable<TResult> RepeatYield<TResult>(TResult element, int co
       this IEnumerable<TSource> source,
       Func<TSource, bool> predicate)
     {
+      CheckNotNull(source, "source");
```

Comment:
Since the use is internal it's tempting to make CheckNotNull debug-only, but there could be some path that delegates its own validation to it.